### ISSUE TITLE
Issues with jest testing

### ISSUE THREAD

**Issue by ThomasDevens** [angular-slider/ngx-slider#222]
I'm having some issues in my test code (Jest) where i'm getting an error within the getLegend function here where it's trying to get a step with an index outside of the steps array (14 steps in the array, with tickStep set to 3). I'm not sure if I have something setup wrong or if there should be an undefined check here. FWIW it seems to work when I test it manually

https://github.com/angular-slider/ngx-slider/blob/e8857e871e75e5a4a48c76b4c9523cee2175fa8b/src/ngx-slider/lib/slider.component.ts#L833

**Comment by piotrdz** [comment]
So there are a few things that are happening here, and this is because of this:
> 14 steps in the array, with tickStep set to 3

I made a mistake in recent release where this would result in one extra tick being created at the end, which would get an out-of-range index. This would result in that bad array access.

Aside from that, the code you pointed to is something I was meaning to rewrite for some time, because it is mixing several different concepts at once - indexes vs values, and possibly overwriting user-specified function.

So I made the change to fix the off-by-one error in creating ticks, and made the `getLegend` code better and safer.

Merged to master, awaiting next release.
